src/GraphQL/Extensions/GraphQLBuilderExtensions.cs

```diff
@@ -1168,5 +1168,28 @@ public static IGraphQLBuilder AddExecutionStrategy<TExecutionStrategy>(this IGra
             return builder;
         }
         #endregion
+
+        #region - UseTelemetry -
+#if NET5_0_OR_GREATER
+        /// <summary>
+        /// Configures the GraphQL server to collect traces via the <see cref="System.Diagnostics.Activity">System.Diagnostics.Activity API</see> and records events that match the
+        /// <see href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/graphql.md">OpenTelemetry recommendations</see>.
+        /// Trace data contains the GraphQL operation name, the operation type, and the optionally the document.
+        /// </summary>
+        /// <remarks>
+        /// When applicable, place after calls to UseAutomaticPersistedQueries to ensure that the query document is recorded properly.
+        /// </remarks>
+        public static IGraphQLBuilder UseTelemetry(this IGraphQLBuilder builder, Action<TelemetryOptions>? configure = null)
+            => UseTelemetry(builder, configure != null ? (opts, _) => configure(opts) : null);
+
+        /// <inheritdoc cref="UseTelemetry(IGraphQLBuilder, Action{TelemetryOptions}?)"/>
+        public static IGraphQLBuilder UseTelemetry(this IGraphQLBuilder builder, Action<TelemetryOptions, IServiceProvider>? configure)
```

Comment:
Let's add overload with generic argument TProvider where TProvider : TelemetryProvider .

Reply:
I would not add additional overloads at this time.  Since we do not know if any options classes are necessary, or which class it would be, similar to `AddErrorInfoProvider<T>()` users would need to register options separately.  And we already have `ConfigureExecution` available for custom implementations so adding another overload serves no real purpose.

Reply:
In a similar fashion, we do not have generic overloads of `UseMemoryCache` or `UseAutomaticPersistedQueries` for the same reason -- `ConfigureExecution` is available for derived classes.

Reply:
I reviewed my decision in favor of delegates in the options so I think overloads will not be needed.